src/test/resources/refaster/EscapesRecipes.java

```diff
@@ -76,8 +91,14 @@ public String getDescription() {
         @Override
         public TreeVisitor<?, ExecutionContext> getVisitor() {
             JavaVisitor<ExecutionContext> javaVisitor = new AbstractRefasterJavaVisitor() {
-                final JavaTemplate before = Semantics.expression(this, "before", (String value) -> String.format("\"%s\"", com.sun.tools.javac.util.Convert.quote(value))).build();
-                final JavaTemplate after = Semantics.expression(this, "after", (String value) -> com.sun.tools.javac.util.Constants.format(value)).build();
+                final JavaTemplate before = JavaTemplate
+                        .builder("String.format(\"\\\"%s\\\"\", com.sun.tools.javac.util.Convert.quote(foo.Escapes.ConstantsFormat.before.value))")
```

Comment:
The `foo.Escapes.ConstantsFormat.before.value` parameter here also looks wrong. Possibly something like this is required? https://github.com/openrewrite/rewrite-templating/blob/b138ac0fa33ccde1fc5305a484bc68b4ab9782d0/src/main/java/org/openrewrite/java/template/processor/TemplateProcessor.java#L97-L101

Reply:
Thanks! Finally managed to get it working, which unblocks running recipes that use Java 11+ constructs on Java 8. 🎉